cmd/sonobuoy/app/gen.go

```diff
@@ -39,21 +42,40 @@ var GenCommand = &cobra.Command{
 }
 
 func init() {
-	args.AddSonobuoyImageFlag(&genopts.Image, GenCommand)
-	args.AddModeFlag(&genopts.ModeName, GenCommand)
-	args.AddNamespaceFlag(&genopts.Namespace, GenCommand)
-
+	AddGenFlags(&genopts, GenCommand)
 	RootCmd.AddCommand(GenCommand)
 }
 
+// AddGenFlags adds generation flags to a command
+func AddGenFlags(gen *ops.GenConfig, cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVarP(
```

Comment:
Do these need to be persistent flags? That to me implies there is a subcommand and this is a global option flag for this command and all children commands.

Reply:
Doesn't hurt if we somehow expanded, but if they were local then we would need to shuffle them all.